app/models/miq_server/worker_management/monitor/systemd.rb

```diff
@@ -29,6 +29,7 @@ def systemd_manager
   def systemd_stop_services(service_names)
     service_names.each do |service_name|
       systemd_manager.StopUnit(service_name, "replace")
+      systemd_manager.ResetFailedUnit(service_name)
```

Comment:
WIP because I want to confirm that this order is correct, (stop, reset, disable)

Reply:
Okay taking out of WIP, tested this live on an appliance by `kill -9`'ing a worker and the failed sytstemd service is properly cleaned up